packages/rich-text/src/plugins/Heading/createHeadingPlugin.ts

```diff
@@ -1,11 +1,23 @@
 import { BLOCKS, HEADINGS } from '@contentful/rich-text-types';
-import { getAbove, onKeyDownToggleElement } from '@udecode/plate-core';
+import { getAbove, toggleNodeType } from '@udecode/plate-core';
+import isHotkey from 'is-hotkey';
 
-import { isInlineOrText } from '../../helpers/editor';
+import { isBlockSelected, isInlineOrText } from '../../helpers/editor';
 import { transformLift, transformUnwrap } from '../../helpers/transformers';
 import { RichTextPlugin } from '../../types';
 import { HeadingComponents } from './components/Heading';
 
+const buildHeadingEventHandler = (type: BLOCKS) => (editor, plugin) => (event) => {
```

Comment:
Please add param types. You may want to use `KeyboardHandler` from `@udecode/plate-core`